src/main/java/net/minecraftforge/items/customslots/IExtensionSlot.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.items.customslots;
+
+import com.google.common.collect.ImmutableSet;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.init.Enchantments;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Describes an extension slot and its contents.
+ */
+public interface IExtensionSlot
+{
+    /**
+     * Gets the container that is managing this slot.
+     *
+     * @return The owning container.
+     */
+    IExtensionContainer getContainer();
+
+    /**
+     * Gets the slot type identifier for this slot.
+     *
+     * @return The resource name serving as a namespaced identifier
+     */
+    ResourceLocation getType();
+
+    /**
+     * Gets the contents of the slot. The stack is *NOT* required to be of an IExtensionSlotItem!
+     *
+     * @return The contained stack
+     */
+    ItemStack getContents();
+
+    /**
+     * Sets the contents of the slot. The stack is *NOT* required to be of an IExtensionSlotItem!
+     *
+     * @param stack The stack to be assigned to the slot
+     */
+    void setContents(ItemStack stack);
+
+    // Permissions
+
+    /**
+     * Queries wether or not the stack can be placed in this slot.
```

Comment:
Change "wether" -> "whether" here and a few other places too.

Reply:
Yay me for copypasting a typo 3 times... :P